pkg/apis/pipeline/v1alpha1/pod_test.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2019 The Tekton Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1_test
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestCombinedPodTemplate(t *testing.T) {
+	affinity := &corev1.Affinity{
+		NodeAffinity: &corev1.NodeAffinity{},
+	}
+	nodeSelector := map[string]string{
+		"banana": "chocolat",
+	}
+	tolerations := []corev1.Toleration{{
+		Key:   "banana",
+		Value: "chocolat",
+	}}
+
+	template := v1alpha1.PodTemplate{
+		NodeSelector: map[string]string{
+			"foo": "bar",
+			"bar": "baz",
+		},
+		Tolerations: []corev1.Toleration{{
+			Key:   "foo",
+			Value: "bar",
+		}},
+		Affinity: &corev1.Affinity{
+			PodAffinity: &corev1.PodAffinity{},
+		},
+	}
+	// Same as template above
```

Comment:
i'm a bit confused! wouldn't we want "banana": "chocolat", etc. to show up in the expected (`got`) template?

Reply:
I should change the names of the variable to make it clearer, good point. So `affinity`, `nodeSelector` and `tolerations` are coming from the deprecated fields, so `PodTemplate` takes precedence over them. I choose to not merge `NodSelector` and `Tolerations` to make it simpler (especially as those deprecated field are going away rather shortly anyway) ; that's why `"banana": "chocolat"` is not in the `got` template :angel:

Reply:
OOOOOOOOH okay i get it, the point is to override them - maybe we should include a deprecated field we don't override as well in one of the tests? (to make sure they can be combined - unless it's supposed to be all or nothing, all depcreated or all new)

Reply:
Ah right, I see what you mean, yeah I'll update the tests